contributors/design-proposals/api-machinery/customresources-validation.md

```diff
@@ -381,6 +381,38 @@ Note:
 
 2. For migration of CRDs with no validation to CRDs with validation, we can create a controller that will validate and annotate invalid CRs once the spec changes, so that the custom controller can choose to delete them (this is also essentially the status condition of the CRD). This can be achieved, but it is not part of the proposal.
 
+### Admission Webhook
+
+Custom resource definitions use the the same REST endpoint as built-in Kubernetes API objects. Consequently standard
+[dynamic web-hook admission controllers](https://kubernetes.io/docs/admin/extensible-admission-controllers/#external-admission-webhooks) 
+can be used for validation of custom resource definitions in addition to built-in resources.
+
+However, because the creation of admission controllers is a fairly high-privilege activity,
+and in many cases the creator of a `CustomResourceDefinition` is third-party extension code
+(e.g. an instance of the _operator pattern_) which should not have the ability to
+create arbitrary admission controllers, it is necessary to have an alternate solution for
+adding web-hook validation to custom resource definitions.
+
+The solution is that the `CustomResourceDefinition` itself has two arrays of webhooks in the
+definition itself. One array will contain non-mutating admission controllers for things like
+validation, and one will have mutating admission controllers for the purposes of defaulting.
+The custom resource controller (a piece of trusted code), will register
+these webhooks, _exclusively_ to be triggered for instances of the specific custom resource.
+This ensures that third-party extension code can not register admission controllers for
+arbitrary API objects.
+
+To achieve this, the `CustomResourceDefinition` has two arrays of `admissionregistration.Webhook` 
```

Comment:
nit: not much new in this paragraph. just move the type `admissionregistration.Webhook` into the previous paragraph.

Reply:
done.